FRAuthenticator/FRAuthenticator/Manager/AuthenticatorManager.swift

```diff
@@ -298,6 +336,53 @@ struct AuthenticatorManager {
     }
     
     
+    /// Lock the given `Account` object limiting the access to all `Mechanism` objects and any `PushNotification` objects associated with it.
+    /// - Parameters:
+    ///     - account: Account object to be locked
+    ///     - policy: The non-compliance policy
+    /// - Throws: AccountError
+    /// - Returns: Boolean result of lock operation
+    @discardableResult public func lockAccount(account: Account, policy: FRAPolicy) throws -> Bool {
+        if account.lock {
+            throw AccountError.failToLockAccountAlreadyLocked
+        } else if policy.name.isEmpty {
+            throw AccountError.failToLockMissingPolicyName
+        } else if !policyEvaluator.isPolicyAttached(account: account, policyName: policy.name) {
+            throw AccountError.failToLockInvalidPolicy
+        } else {
+            account.lock(policy: policy)
+            return storageClient.setAccount(account: account)
+        }
+    }
+    
+    
+    /// Unlock the  given `Account` object
+    /// - Parameter account: Account object to be locked
+    /// - Throws: AccountError
+    /// - Returns: Boolean result of unlock operation
+    @discardableResult public func unlockAccount(account: Account) throws -> Bool {
+        if !account.lock {
+            throw AccountError.failToUnlockAccountNotLocked
+        } else {
+            account.unlock()
+            return storageClient.setAccount(account: account)
+        }
+    }
+    
+    private func evaluatePoliciesForAccount(account: Account) -> Void {
+        let result = self.policyEvaluator.evaluate(account: account)
+        if(!result.comply) {
+            FRALog.w("Locking Account ID (\(account.identifier)) due non-compliance policy: \(result.nonCompliancePolicy!.name)")
+            account.lock(policy: result.nonCompliancePolicy!)
```

Comment:
Same here. Can we avoid force unwrapping just to be safe? ?? default value would be preferred

Reply:
The policy is mandatory, default value is not an option. I will use another approach to be safe.